contrib/epee/src/net_utils_base.cpp

```diff
@@ -84,5 +68,31 @@ namespace epee { namespace net_utils
     return ss.str();
   }
 
+  const char* zone_to_string(zone value) noexcept
+  {
+    switch (value)
+    {
+    case zone::kPublic:
+      return "public";
+    case zone::kI2P:
+      return "i2p";
+    case zone::kTor:
+      return "tor";
+    default:
+      break;
+    }
+    return "invalid";
+  }
+
+  zone zone_from_string(const boost::string_ref value) noexcept
```

Comment:
Minor: this will be coming in from `args` so should there be any formatting cleanup?

Reply:
Do you mean for case insensitive, etc? I thought about it, but it didn't seem like it would ruin anything to have strict checking. Is there some other cleanup I've missed?